### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by daun**
- Some ideas for the readme
- I've added variants for the `rules` section: table, list, and headings, not sure here but I prefer the more compact list version because it corresponds directly to the example code above: 1 line per property
- Best to view this from the branch's [readme view](https://github.com/swup/fragment-plugin/tree/docs#readme)
- Includes two drive-by improvements
  - extraction of default options into `this.defaults`
  - removal of unused dependencies

### CODE REVIEW

**Comment by daun** [comment]
Ready for review :)

File: README.md [hunk 1]
```diff
@@ -124,58 +136,137 @@ export type PluginOptions = {
...
+`name` of this rule to allow precise styling. The from/to patterns are converted to regular
+expressions by [path-to-regexp](https://www.npmjs.com/package/path-to-regexp).
+
+```js
+{
+  rules: [
+    {
+      from: '/users/:filter?',
+      to: '/users/:filter?',
+      fragments: ['#users'],
```
Comment (hirasso) [inline README.md:150]: Even though regular expressions are supported, I would not advertise them. Using strings or arrays of strings and let path-to-regex do the heavy lifting will prevent many problems.

File: README.md [hunk 2]
```diff
@@ -124,58 +136,137 @@ export type PluginOptions = {
...
+      name: 'list'
+    }
+  ]
+}
+```
+
+### (List)
 
-#### from (required)
+- `from`, required, `string | string[]` — The pattern(s) to match against the previous URL
```
Comment (hirasso) [inline README.md:159]: Two times "optional"

File: README.md [hunk 3]
```diff
@@ -124,58 +136,137 @@ export type PluginOptions = {
 
 ### rules
 
-An array of rules consisting of these properties:
+The rules that define whether a visit will be considered a fragment visit. Each rule consists of
+mandatory `from` and `to` URL patterns, an array `fragments` of selectors, as well as an optional
+`name` of this rule to allow precise styling. The from/to patterns are converted to regular
```
Comment (hirasso) [inline README.md:141]: I would stick to "scoped" styling, as repeated further down

File: src/SwupFragmentPlugin.ts [hunk 4]
```diff
@@ -58,22 +58,20 @@ export default class SwupFragmentPlugin extends PluginBase {
 
 	logger: Logger;
 
+	defaults: PluginOptions = {
```
Comment (hirasso) [inline src/SwupFragmentPlugin.ts:61]: Why is this important to you?

**Review by hirasso — COMMENTED** [review]
Very nice work! 🤸‍♀️

File: README.md [hunk 5]
```diff
@@ -106,15 +119,13 @@ html.is-animating .transition-main {
...
-
-## Plugin Options
+## Options
 
 ```typescript
 export type PluginOptions = {
   rules: Array<{
     from: string | string[];
-    to: string | string[];
+    from: string | string[];
```
[the remark this answers is not in the document]
Comment (daun) [reply README.md]: Pushed a fix :)

File: README.md [hunk 6]
```diff
@@ -1,102 +1,115 @@
...
   transition: opacity 250ms;
   opacity: 1;
 }
 html.is-animating .transition-main {
   opacity: 0;
 }
+
 /*
-* The animation when filtering users
+* The animation for the fragment rule named "users"
```
[the remark this answers is not in the document]
Comment (daun) [reply README.md:113]: Sounds good!

File: README.md (hunk 2 above)
Comment (daun) [reply README.md:159]: Not sure what this is referring to? Maybe wrong line marked?

File: README.md (hunk 1 above)
Comment (daun) [reply README.md:150]: Updated to: 'The from/to patterns are interpreted and follow the syntax established by path-to-regexp'. Does that make sense?

File: src/SwupFragmentPlugin.ts (hunk 4 above)
Comment (daun) [reply src/SwupFragmentPlugin.ts:61]: When overwriting or manipulating the plugin options from the outside, it's great to have access to the default or fallback version of them. Not sure why I give so much of a damn about this to be honest, haha. It's become almost philosophical at this point.

**Comment by daun** [comment]
Pushed a commit with the suggested updates/reversal.

**Comment by daun** [comment]
Feel free to accept as much or little of this PR as you like! This is your baby :) I do feel that some things are easier for me to express as a "user", i.e. as somebody unfamiliar with the exact workings of the plugin. But yeah, it's all suggestions!

**Comment by hirasso** [comment]
Regarding the rules presentation, I like both the list as well as the table better then the headings. You decide! :)

**Comment by hirasso** [comment]
Woaaa, just saw your explanations for the attributes. Amazing work!!! 🏖️🏕️💥

File: README.md (hunk 2 above)
Comment (hirasso) [reply README.md]: I think this was already fixed in one of your commits

**Pull request merged**